app/src/main/java/com/concordium/wallet/ui/cis2/lookfornew/TokenDetailsFragment.kt

```diff
@@ -171,16 +169,55 @@ class TokenDetailsFragment : TokensBaseFragment() {
     }
 
     private fun showMetadata(tokenMetadata: TokenMetadata) {
```

Comment:
Instead of having this logic twice, and since on the naming I expect we are migrating to `lookfornew` views what about making a static method like this

```
        @JvmStatic
        fun setMetaData(
            tokenMetadata: TokenMetadata,
            dialogRoot: androidx.constraintlayout.widget.ConstraintLayout,
            dialogBinding: com.concordium.wallet.databinding.ShowRawMetadataDialogBinding,
            layout: androidx.appcompat.widget.LinearLayoutCompat
        ) {
            with(dialogRoot) {
                visibility = View.GONE
                setOnClickListener { visibility = View.GONE }
            }

            dialogBinding.closeDialog.setOnClickListener { dialogRoot.visibility = View.GONE }

            layout.setOnClickListener {
                dialogBinding.apply {
                    thumbnailTitle.visibility = if (tokenMetadata.thumbnail != null) View.VISIBLE else View.GONE
                    thumbnailValue.visibility = if (tokenMetadata.thumbnail != null) View.VISIBLE else View.GONE
                    tokenMetadata.thumbnail?.let { thumbnail -> thumbnailValue.text = thumbnail.url }

                    displayTitle.visibility = if (tokenMetadata.display != null) View.VISIBLE else View.GONE
                    displayValue.visibility = if (tokenMetadata.display != null) View.VISIBLE else View.GONE
                    tokenMetadata.display?.let { display -> displayValue.text = display.url }

                    nameValue.text = tokenMetadata.name
                    decimalValue.text = tokenMetadata.decimals.toString()
                    descriptionValue.text = tokenMetadata.description

                    symbolTitle.visibility = if (tokenMetadata.symbol != null) View.VISIBLE else View.GONE
                    symbolValue.visibility = if (tokenMetadata.symbol != null) View.VISIBLE else View.GONE
                    tokenMetadata.symbol?.let { symbol -> symbolValue.text = symbol }

                    uniqueValue.text = tokenMetadata.unique.toString()
                }

                dialogRoot.visibility = View.VISIBLE
            }
        }
```

and the both in `TokenDetailsFragment` and `TokenDetailsActivity` call this like

```
    private fun showMetadata(tokenMetadata: TokenMetadata) {
        val dialogRoot = binding.showRawMetadataDialogRoot;
        val dialogBinding = binding.showRawMetadataDialogContainer;
        val layout = binding.details.showRawMetadataHolder;

        setMetaData(tokenMetadata, dialogRoot, dialogBinding, layout);
    }
```

and

```
    private fun showMetadata(tokenMetadata: TokenMetadata) {
        val dialogRoot = binding.showRawMetadataDialogRoot;
        val dialogBinding = binding.showRawMetadataDialogContainer;
        val layout = binding.includeAbout.showRawMetadataHolder;

        TokenDetailsFragment.setMetaData(tokenMetadata, dialogRoot, dialogBinding, layout);
    }
```

Reply:
nice